src/workerd/api/streams/standard.c++

```diff
@@ -524,22 +524,26 @@ int getHighWaterMark(const UnderlyingSource& underlyingSource,
 
 // It is possible for the controller state to be released synchronously while
 // we are in the middle of a read. When that happens we need to defer the actual
-// close/error state change until the read call is complete. ReadPendingScope
-// handles this for us. It is constructed when we start a read and destructed
-// synchronously when the call to read completes.
-template <typename C>
-struct ReadPendingScope {
-  jsg::Lock& js;
-  C& controller;
-  ReadPendingScope(jsg::Lock& js, C& controller)
-      : js(js), controller(controller) {
+// close/error state change until the read call is complete. readPendingScope
+// handles this for us.
```

Comment:
this is a tricky code, I would very much like this sentence to continue, e.g.: `handles this for us by making sure that pendingReadCount is incremented until the read operation completes and deferring a state change until it is 0`. (did I get the code right btw?)